testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/AggregateRealmTestCase.java

```diff
@@ -506,176 +614,227 @@ private void assertNotInRole(final String rolePrintResponse, String role) {
+                            .build())
+                    .build());
+            configurableElements.add(UndertowApplicationSecurityDomain.builder()
+                              .withName(AGGREGATE_REALM_SAME_TYPE_NAME)
+                    .withSecurityDomain(AGGREGATE_REALM_SAME_TYPE_NAME)
+                    .build());
+            configurableElements.add(UndertowApplicationSecurityDomain.builder()
+                              .withName(AGGREGATE_REALM_DIFFERENT_TYPE_NAME)
+                    .withSecurityDomain(AGGREGATE_REALM_DIFFERENT_TYPE_NAME)
+                    .build());
+
+            configurableElements.add(new CustomFSAttributes(FILESYSTEM_REALM_AUTHN_NAME, USER_FIRST_ATTRIBUTES, "Colours", "Red", "Orange"));
+            configurableElements.add(new CustomFSAttributes(FILESYSTEM_REALM_2_AUTHN_NAME, USER_SECOND_ATTRIBUTES, "Colours", "Yellow", "Green"));
+            configurableElements.add(new CustomFSAttributes(FILESYSTEM_REALM_AUTHN_NAME, USER_COMBINED_ATTRIBUTES, "Colours", "Blue", "Violet"));
+            configurableElements.add(new CustomFSAttributes(FILESYSTEM_REALM_AUTHN_NAME, USER_COMBINED_ATTRIBUTES, "Year", "1979"));
+            configurableElements.add(new CustomFSAttributes(FILESYSTEM_REALM_2_AUTHN_NAME, USER_COMBINED_ATTRIBUTES, "Colours", "Pink", "Turqoise"));
+            configurableElements.add(new CustomFSAttributes(FILESYSTEM_REALM_2_AUTHN_NAME, USER_COMBINED_ATTRIBUTES, "City", "San Francisco"));
+
+            configurableElements.add(AggregateSecurityRealm.builder(AGGREGATE_ATTRIBUTES_NAME)
+                    .withAuthenticationRealm(PROPERTIES_REALM_AUTHN_NAME)
+                    .withAuthorizationRealms(FILESYSTEM_REALM_AUTHN_NAME, FILESYSTEM_REALM_2_AUTHN_NAME)
+                    .build());
+            configurableElements.add(SimpleSecurityDomain.builder()
+                    .withName(AGGREGATE_ATTRIBUTES_NAME)
+                    .withDefaultRealm(AGGREGATE_ATTRIBUTES_NAME)
+                    .withPermissionMapper("default-permission-mapper")
+                    .withRealms(SimpleSecurityDomain.SecurityDomainRealm.builder()
+                            .withRealm(AGGREGATE_ATTRIBUTES_NAME)
+                            .withRoleDecoder("groups-to-roles")
+                            .build())
+                    .build());
+            configurableElements.add(UndertowApplicationSecurityDomain.builder()
+                    .withName(AGGREGATE_ATTRIBUTES_NAME)
+                    .withSecurityDomain(AGGREGATE_ATTRIBUTES_NAME)
+                    .build());
+
+            return configurableElements.toArray(new ConfigurableElement[configurableElements.size()]);
         }
+    }
 
-        private String createPropertiesUserWithHashedPassword(String username, String password, String realmName) {
-            return username + "=" + createHashedPassword(username, password, realmName) + "\n";
-        }
+     static class CustomFSAttributes implements ConfigurableElement {
```

Comment:
CustomFSAttributes class (implementation of ConfigurableElement) needs to override remove method from ConfigurableElement element. If remove method won't be override, [default remove method will be used](https://github.com/wildfly/wildfly/blob/master/testsuite/shared/src/main/java/org/wildfly/test/security/common/elytron/ConfigurableElement.java#L62) and it leads to IllegalStateException during test cleanup and cleanup won't be finished. Many elements won't be removed and could cause errors of other tests. This could leads to many false-positive errors during the CI runs.

Reply:
Good catch, thank you.